src/frequenz/sdk/actor/power_distributing/power_distributing.py

```diff
@@ -132,8 +132,32 @@ async def _run(self) -> None:  # pylint: disable=too-many-locals
         """
         await self._component_manager.start()
 
+        power_tasks: dict[str, asyncio.Task[None]] = {}
+
+        def cleanup(task: asyncio.Task[None]) -> None:
+            task_name = task.get_name()
+            assert task_name in power_tasks, "Task not found in power_tasks"
+
+            # Raise if there was an exception while processing the task
+            task.result()
+
+            del power_tasks[task_name]
+
         async for request in self._requests_receiver:
-            await self._component_manager.distribute_power(request)
+            task_name = f"power_request_ids={request.component_ids}"
```

Comment:
Maybe add more info, like a prefix with `{type(self).__name__}:`.

Reply:
makes sense. I was also considering to hash the list component IDs as the list could be a long list of components

Reply:
Updated as suggested